draft-ietf-lwig-minimal-esp.xml

```diff
@@ -292,52 +292,49 @@ As a result, it should not be part of a minimal implementation of ESP.  </t>
-In addition, some devices may also embed radio modules with hardware acceleration for AES-CCM, in which case, this mode may be preferred.</t>
-
-<t> Power Consumption and Bandwidth Consumption: Similarly to the encryption algorithm transform complexity, reducing the payload sent, may significantly reduce the energy consumption of the device. 
-As a result, encryption algorithm transforms with low overhead may be considered. 
+
+<t> Security: Security is the criteria that should be considered first for the selection of encryption algorithm transform.
+The security of encryption algorithm transforms is expected to evolve over time, and it is of primary importance to follow up-to-date security guidance and recommendations.
+The chosen encryption algorithm must not be vulnerable or weak (see <xref target="RFC8221"/> for outdated ciphers).
+ESP can be used to authenticate only (ENCR_NULL) or to encrypt the communication.
+In the latter case, authenticated encryption (AEAD) is RECOMMENDED <xref target="RFC8221"/>.</t>
+
+<t>Resilience to nonce re-use: Some transforms -including AES-GCM - are vulnerable to nonce collision with a given key.
+While the generation of the nonce may prevent such collision during a session, the mechanisms are unlikely to provide such protection across sleep states or reboot.
+This causes an issue for devices that are configured using static keys (called manual keying) and manual keying should not be used with these encryption algorithms.
+When the key is likely to be re-used across reboots, algorithms that are nonce misuse resistant such as, for example, AES-SIV <xref target="RFC5297"/>, AES-GCM-SIV <xref target="RFC8452"/> or Deoxys-II <xref target="DeoxysII"/> are RECOMMENDED.
+Note however that currently none of these are yet defined for use with ESP.
+[Paul: missing guidance on what to do then for devices using manual keying?]
+</t>
+
+<t> Interoperability: constrained devices usually only implement one or very few different encryption algorithm transforms.
+<xref target="RFC8221"/> takes the life cycle of encryption algorithm transforms and device manufactoring into consideration in its recommendations for mandatory-to-implement ("MTI") algorithms.
+</t>
+
+<t> Power Consumption and Cipher Suite Complexity: Complexity of the encryption algorithm transform and the energy cost associated with it are especially important considerations for devices that have limited resources or are battery powered.
+The battery life might determine the lifetime of the entire device.
+The choice of a cryptographic function should consider re-using specific libraries or to take advantage of hardware acceleration provided by the device.
+For example, if the device benefits from AES hardware modules and uses AES-CTR, it may prefer ENCR_AES_CBC [Paul: not ENCR_AES_CTR??] for its encryption.
+In addition, some devices may also embed radio modules with hardware acceleration for AES-CCM, in which case, this transform may be preferred.</t>
```

Comment:
ENCR_AES_CBC [Paul: not ENCR_AES_CTR??] for its encryption.

needs to be replaced by: 
AUTH_AES-XCBC for its authentication.

AES-CTR needs authentication.

Reply:
We were talking about how some AES hardware could enhance some AES algorithms. I just changed the names to the ones that are actually used in the IANA Registry for IKEv2. There is no need to talk about AEAD vs ENCR+AUTH. We are talking about acceleration of components here.

Reply:
if you use AES-CTR you may prefer AUTH_AES-XCB that SHA2 based cipher suites to take advantage of the acceleration of components.

Reply:
Sure, good example. My only point was that you should refer to the hardware algo by its common name (like AES) and you should refer to the ESP/IKE algorithms by their official IANA names (eg ENCR_AES_CTR and AUTH_AES-XCBC)